### ISSUE TITLE
itemId can't contain word separator

### ISSUE THREAD

**Issue by soukicz** [PredictionIO/template-scala-parallel-universal-recommendation#12]
It took me couple hours to figure out, but item ID can't contain dash (or other separator) - for example _1-123_

If it contains dash, query is always returning zero scores. It is probable bug in elastic configuration - analyzer should by _keyword_ and not _standard_

**Comment by pferrel** [comment]
v 0.1.1 should have fixed that. We recommend you update the v0.2.1, which is the latest.

**Comment by soukicz** [comment]
I was using 0.2.0

**Comment by pferrel** [comment]
ok, I'll look. All fields are not_analyzed in Elasticsearch.

**Comment by pferrel** [comment]
I changed the sample data to have spaces and hyphens in user and item ids but it seems to work fine. Can you give some sample data that fails for you?

**Comment by pferrel** [comment]
some samples queries that illustrate the ids I tried, which all return the correct results. 

```
echo ""
echo "Recommendations for user: u1"
echo ""
curl -H "Content-Type: application/json" -d '
{
    "user": "u1"
}' http://localhost:8000/queries.json
echo ""

echo ""
echo "Recommendations for user: u 2"
echo ""
curl -H "Content-Type: application/json" -d '
{
    "user": "u 2"
}' http://localhost:8000/queries.json
echo ""

echo ""
echo "Recommendations for user: u-3"
echo ""
curl -H "Content-Type: application/json" -d '
{
    "user": "u-3"
}' http://localhost:8000/queries.json
echo ""

echo ""
echo "Recommendations for user: ipad-retina"
echo ""
curl -H "Content-Type: application/json" -d '
{
    "item": "ipad-retina"
}' http://localhost:8000/queries.json
echo ""

echo ""
echo "Recommendations for user: iphone 4"
echo ""
curl -H "Content-Type: application/json" -d '
{
    "item": "iphone 4"
}' http://localhost:8000/queries.json
echo ""
```

**Comment by pferrel** [comment]
There was a regression in v0.2.0 when the index hot-swap was added.
It has been corrected in v0.2.2
